ako-gateway-api/k8s/validator.go

```diff
@@ -145,6 +157,26 @@ func isValidListener(key string, gateway *gatewayv1.Gateway, gatewayStatus *gate
 		return false
 	}
 
+	// hostname should not overlap with hostname of an existing gateway
+	gatewayNsList, err := akogatewayapilib.AKOControlConfig().GatewayApiInformers().GatewayInformer.Lister().Gateways(gateway.Namespace).List(labels.Set(nil).AsSelector())
```

Comment:
can we add this code during ingestion of Gateway `IsValidGateway`?

Reply:
This function `isValidListener` is called from within `IsValidGateway` only.
Did you mean to say moving _**line 161 only**_ somewhere up within IsValidGateway ?